WcaOnRails/app/controllers/users_controller.rb

```diff
@@ -69,13 +69,13 @@ def select_nearby_delegate
 
   def edit_avatar_thumbnail
     @user = user_to_edit
-    return if redirect_if_cannot_edit_user(@user)
+    return if redirect_unless_user(:can_change_users_avatar?, @user)
```

Comment:
(Sorry for insta-merging this PR before you got a chance to comment.)

`redirect_unless_user` checks a condition, and if the user doesn't meet the criteria, it redirects to the home page with an alert telling the user what it is they could not do. See code [here](https://github.com/jfly/worldcubeassociation.org/blob/96ef7ba9eeb162995b167fd15f04d50f1cf8deb3/WcaOnRails/app/controllers/application_controller.rb#L76-L85).

After redirecting, we don't want any more code of the controller action to run, which is why `redirect_unelss_user` returns a boolean indicating if a redirect occurred, so we can conditionally return and bail out of the controller action. You are correct that returning at the end of the function is pointless, but it's very possible that these `return if ...` lines have been moved/copied around and weren't always the last thing in a function. I personally like the simplicity of saying "anytime you call redirect_unless_user, you need to conditionally return".

(There may be a better way of handling all of this, so I'm very open to suggestions)